api/paths/paths.go

```diff
@@ -1,80 +1,138 @@
-package types
+package paths
 
 import (
 	"bytes"
 	"fmt"
 	"io"
+	"log"
 	"os"
+	"os/user"
 	"path"
 	"regexp"
 	"runtime"
+	"strconv"
+	"strings"
 	"sync"
 	"time"
-
-	log "github.com/Sirupsen/logrus"
-	"github.com/akutz/gotil"
 )
 
 var (
-	libstorageHome = os.Getenv("LIBSTORAGE_HOME")
-	fileKeyCache   = map[fileKey]string{}
-	homeLock       = &sync.Mutex{}
-	thisExeDir     string
-	thisExeName    string
-	thisExeAbsPath string
-	slashRX        = regexp.MustCompile(`^((?:/)|(?:[a-zA-Z]\:\\?))?$`)
-
-	etcEnvVarPath string
-	libEnvVarPath string
-	logEnvVarPath string
-	runEnvVarPath string
-	tlsEnvVarPath string
-	lsxEnvVarPath string
+	debug, _ = strconv.ParseBool(os.Getenv("LIBSTORAGE_DEBUG"))
+
+	pkgLock      = &sync.RWMutex{}
+	homeLock     = &sync.Mutex{}
+	fileKeyCache = map[fileKey]string{}
+	slashRX      = regexp.MustCompile(`^((?:/)|(?:[a-zA-Z]\:\\?))?$`)
+
+	appHome        string
+	appHomeEtcPath string
+	appHomeLibPath string
+	appHomeLogPath string
+	appHomeRunPath string
+	appHomeTLSPath string
+	appHomeLSXPath string
+
+	envVarHome       = "LIBSTORAGE_HOME"
+	envVarHomeEtc    = "LIBSTORAGE_HOME_ETC"
+	envVarHomeEtcTLS = "LIBSTORAGE_HOME_ETC_TLS"
+	envVarHomeLib    = "LIBSTORAGE_HOME_LIB"
+	envVarHomeLog    = "LIBSTORAGE_HOME_LOG"
+	envVarHomeRun    = "LIBSTORAGE_HOME_RUN"
+	envVarHomeLSX    = "LIBSTORAGE_HOME_LSX"
 )
 
 func init() {
+	monFunc := func(pEV, pV *string) {
+		if v := os.Getenv(*pEV); v != "" && v != *pV {
+			Init(appToken)
+		}
+	}
+	monLoop := func(pEV, pV *string) {
+		go func() {
+			for {
+				monFunc(pEV, pV)
+			}
+		}()
+	}
+
+	monLoop(&envVarHome, &appHome)
+	monLoop(&envVarHomeEtc, &appHomeEtcPath)
+	monLoop(&envVarHomeEtcTLS, &appHomeTLSPath)
+	monLoop(&envVarHomeLib, &appHomeLibPath)
+	monLoop(&envVarHomeLog, &appHomeLogPath)
+	monLoop(&envVarHomeRun, &appHomeRunPath)
+	monLoop(&envVarHomeLSX, &appHomeLSXPath)
+}
+
+const (
+	defaultAppToken = "libstorage"
+)
+
+var appToken = defaultAppToken
```

Comment:
Would you be opposed to moving this `const` and `var` definition above `init`? I like to see them all up at the top, but even more, when I read through `init` my first thought was, "but where was `appToken` defined" ?